### PR TITLE
BKNDLSS-31618 [MapBox Component] On mobile devices display directions under the map

### PR DESCRIPTION
**Pull request by addKiwi**

### CODE REVIEW

File: components/bl-mapbox-component/src/helpers.js [hunk 1]
```diff
@@ -96,12 +96,19 @@ export const initMapboxLibrary = (mapRef, mapContainerRef, component, eventHandl
...
 
   if (directions) {
-    map.addControl(
-      new MapboxDirections({
-        accessToken: mapboxgl.accessToken,
-      }),
-      'top-left'
-    );
+    const directions = new MapboxDirections({
+      accessToken: mapboxgl.accessToken,
```
Comment (v-excelsior) [inline components/bl-mapbox-component/src/helpers.js:100]: Could we get all needed in this function parts of mapboxgl using destruction?

`const {Map, accessToken, FullscreenControl ... } = Mapbox`


Also change it

`import mapboxgl from './lib/mapbox';` => `import Mapbox from './lib/mapbox';`

File: components/bl-mapbox-component/src/helpers.js [hunk 2]
```diff
@@ -289,3 +296,11 @@ export const useGeolocation = (mapRef, onDeterminingGeoposition, map) => {
...
+  })(navigator.userAgent || navigator.vendor || window.opera);
```
Comment (v-excelsior) [inline components/bl-mapbox-component/src/helpers.js:304]: `return /long rexexp/i.test(a.substr(0, 4))`

I'm think we doesn't need to IIFE here

**Review by v-excelsior — CHANGES_REQUESTED** [review]

File: components/bl-mapbox-component/src/helpers.js [hunk 3]
```diff
@@ -77,15 +77,25 @@ export const applyFog = (mapRef, component, map) => {
   });
 };
 
+const appendMobileMark = (directions, mapRef, mapContainerRef) => {
```
[the remark this answers is not in the document]
Comment (addKiwi) [reply components/bl-mapbox-component/src/helpers.js:80]: image

File: components/bl-mapbox-component/src/helpers.js [hunk 4]
```diff
@@ -96,12 +106,14 @@ export const initMapboxLibrary = (mapRef, mapContainerRef, component, eventHandl
...
 
   if (directions) {
-    map.addControl(
-      new MapboxDirections({
-        accessToken: mapboxgl.accessToken,
-      }),
-      'top-left'
-    );
+    const directions = new MapboxDirections({ accessToken });
+    const isFit = mapRef.current.transform.width > 570;
```
[the remark this answers is not in the document]
Comment (addKiwi) [reply components/bl-mapbox-component/src/helpers.js:110]: No, this only works for the first load. To make it responsive, I'm not sure I'll have time to finish and release it by the end of the day. Talked to Bohdan, it's fine and we will update it after the release.

**Pull request merged**